### PR TITLE
4295: DoD, ux updates

### PR DESCRIPTION
**Pull request by kkoskelin**
Screen Shot 2020-07-13 at 12 45 52 PM
conditional around persistence method to prevent interactor error in persistence;
un-backgrounding about half the calls for cognito account creation to prevent "Rate limit exceeded" and allow for all the accounts to actually be created (could be suppressed by rate limit error);
package updates;
updating validation for specific case in which the internal user may want to save an unfinished/unserved docket entry that does not yet have a file uploaded;

### CODE REVIEW

File: web-client/src/presenter/actions/DocketEntry/suggestSaveForLaterValidation.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const suggestSaveForLaterValidationAction = ({ props }) => {
```
Comment (sutt0n) [inline web-client/src/presenter/actions/DocketEntry/suggestSaveForLaterValidation.js]: nits:
- file name to append `Action` at the end
- unit tests

File: shared/src/business/useCases/docketEntry/updateDocketEntryInteractor.js [hunk 2]
```diff
@@ -111,11 +111,12 @@ exports.updateDocketEntryInteractor = async ({
     const workItemToDelete = currentDocument.workItems.find(
       workItem => !workItem.document.isFileAttached,
     );
-
-    await applicationContext.getPersistenceGateway().deleteWorkItemFromInbox({
-      applicationContext,
-      workItem: workItemToDelete,
-    });
+    if (workItemToDelete) {
```
Comment (sutt0n) [inline shared/src/business/useCases/docketEntry/updateDocketEntryInteractor.js:118]: nit: probably need a unit test case for this

File: web-client/integration-tests/journey/docketClerkAddsDocketEntries.js [hunk 3]
```diff
@@ -21,12 +21,13 @@ export const docketClerkAddsDocketEntries = (test, fakeFile) => {
...
 
-    expect(test.getState('validationErrors')).toEqual({
+    expect(test.getState('validationErrors')).toMatchObject({
       dateReceived: VALIDATION_ERROR_MESSAGES.dateReceived[1],
       documentType: VALIDATION_ERROR_MESSAGES.documentType[1],
       eventCode: VALIDATION_ERROR_MESSAGES.eventCode,
       partyPrimary: VALIDATION_ERROR_MESSAGES.partyPrimary,
-      primaryDocumentFile: VALIDATION_ERROR_MESSAGES.primaryDocumentFile,
+      primaryDocumentFile:
+        'Scan or upload a document to serve, or click Save for Later to serve at a later time', // matches anything but null or undefined
```
Comment (rachaelparris) [inline web-client/integration-tests/journey/docketClerkAddsDocketEntries.js]: Could you add some clarification about this comment? I'm not following.

**Review by rachaelparris — APPROVED** [review]

**Review by sutt0n — APPROVED** [review]

File: web-client/integration-tests/journey/docketClerkAddsDocketEntries.js (hunk 3 above)
Comment (kkoskelin) [reply web-client/integration-tests/journey/docketClerkAddsDocketEntries.js]: I changed the 'expect', but not the comment. my bad, it will be removed.

**Pull request merged**